tests/test_forward_ref.py

```diff
@@ -1302,22 +1302,20 @@ def inner():
         func_val(a=1)
 
 
-@pytest.mark.xfail(
-    reason='In `GenerateSchema`, only the current class module is taken into account. '
-    'This is similar to `test_uses_the_correct_globals_to_resolve_model_forward_refs`.'
-)
```

Comment:
Turns out this used to work. However, the mentioned `test_uses_the_correct_globals_to_resolve_model_forward_refs` test was also failing on 2.9